src/geojson/geojson.ml

```diff
@@ -296,10 +296,11 @@ module Make (J : Geojson_intf.Json) = struct
               "A Geojson feature requires the type `Feature`. No type was \
                found.")
 
-    let to_json (t, props) =
+    let to_json (id, t, props) =
```

Comment:
You will need to change the `type t` to match this and as @streetCoderr points out, it might be better if we starting using records instead of tuples here. Concretely, I think we should now have:

```ocaml
type id = [ `String of string | `Int of int ] 

type t = {
  id : id option;
  geometry : Geometry.t option;
  properties : json option
}
```

Note we've created our own type for the identifier matching the RFC:

> Feature object with the name "id", and the value of this member is either a JSON string or number.